Week1

jessiwitti opened the pull request:
R package with one initial function "compare_words".

jessiwitti committed:
Added R Skeleton

Added R skeleton with the compare words function (not sure whether I will keep this function in the end or whether I can make this easier)

jessiwitti committed:
Added a description

Added a help file

jessiwitti committed:
Version with example in it

A helpfile version with example in it

jessiwitti committed:
Examples Work

examples should work now

brunykrijgsman commented on GermanWordle/R/Week2functions.R:
```diff
@@ -0,0 +1,41 @@
+#' @param target_str character. Should be a word with the same length as the guessed word
+#' @param guess_str character. Should be a word with the same length as the target word
+#' 
+#' @return Shows whether the letter is in the word and on the correct place, in the word but incorrect place, or not in the word
```
Suggestion: Add the return type (array of strings?).

brunykrijgsman reviewed (approved):
- The R package skeleton looks good. Suggestion: You could add some things to your .gitignore (e.g., .DS-Store and .Rhistory).
- Clear function documentation (I made one suggestion in the code).

bdenizs commented on GermanWordle/R/Week2functions.R:
```diff
@@ -0,0 +1,41 @@
+  target <- strsplit(target_str, "")[[1]]
+  guess <- strsplit(guess_str, "")[[1]]
+  result <- character(nchar(guess_str))
+  
```
Just a suggestion but you can add comments to your function and elaborate a bit more, and also check whether the word the users guessed is in the wordlist.

akane7011 replied:
It would be better to turn the input into lowercase letters - so that the capitalization doesn't matter here.

bdenizs reviewed (commented):
In general I liked your wordle game. Your description was brief but explanatory as it's a very well-known game. Your R skeleton was complete and you did a good job on documentation, I just agree with the other review that you should add the return type. I also added a suggestion for the code :)

akane7011 reviewed (commented):
The code looks concise and clear to me! 
Also depending on the rules, you may want to print the word length at the start of the game (or fixed to 5, for example).

jessiwitti committed:
Update .gitignore

Added .DS_Store & .Rhistory to gitignore

jessiwitti committed:
Included feedback

Generate a target word, compare the guessed word with the target word, added comments + that the results are a character vector
